lib/qrcode.js

```diff
@@ -0,0 +1,152 @@
+const QRCodeStyling = require("qr-code-styling");
+const {JSDOM} = require("jsdom");
+const nodeCanvas = require("canvas");
+
+/**
+ * This is https://wiki.nottinghack.org.uk/logo/nottinghack_with_white.svg as a png with base64 encoding,
+ * to avoid downloading it each time, or needing to add to the docker image.
+ */
+const nhLogoBase64 = `iVBORw0KGgoAAAANSUhEUgAAAIcAAACHCAYAAAA850oKAAAACXBIWXMAAAAAAAAAAQCEeRdzAAAQ
```

Comment:
I know this looks like a remote code execution attempt, but it's not. It's just the easiest way to make a small image available to the application.